templates/vue/src/components/modals/NodeModal/forms/ContentForm/MultipleChoiceForm.vue

```diff
@@ -0,0 +1,274 @@
+    },
+    nextChoiceRowRadioId(newNextChoiceRowRadioId) {
+      this.question.answerTypes.multipleChoice.nextChoiceRowRadioId = newNextChoiceRowRadioId
+    },
+  },
+  created() {
+    if (
+      !this.question.answerTypes.multipleChoice.hasOwnProperty("checkboxArray") ||
+      !this.question.answerTypes.multipleChoice.hasOwnProperty("radioArray") ||
+      !this.question.answerTypes.multipleChoice.hasOwnProperty("useImages")
+    ) {
+      this.question.answerTypes.multipleChoice.checkboxArray = this.choiceRows
+      this.question.answerTypes.multipleChoice.radioArray = this.choiceRowsRadio
+      this.question.answerTypes.multipleChoice.nextChoiceRowId = this.nextChoiceRowId
+      this.question.answerTypes.multipleChoice.nextChoiceRowRadioId = this.nextChoiceRowRadioId
+      this.question.answerTypes.multipleChoice.useImages = this.useImages
+      this.question.answerTypes.multipleChoice.preSelectedCheckBoxOptions = this.preSelectedCheckBoxOptions
+      this.question.answerTypes.multipleChoice.preSelectedRadioOptions = this.preSelectedRadioOptions
+    } else {
+      this.choiceRows = this.question.answerTypes.multipleChoice.checkboxArray
+      this.choiceRowsRadio = this.question.answerTypes.multipleChoice.radioArray
+      this.nextChoiceRowId = this.question.answerTypes.multipleChoice.nextChoiceRowId
+      this.nextChoiceRowRadioId = this.question.answerTypes.multipleChoice.nextChoiceRowRadioId
+      this.useImages = this.question.answerTypes.multipleChoice.useImages
+      this.preSelectedCheckBoxOptions = this.question.answerTypes.multipleChoice.preSelectedCheckBoxOptions
+      this.preSelectedRadioOptions = this.question.answerTypes.multipleChoice.preSelectedRadioOptions
+    }
+  },
+  methods: {
+    addNewChoice: function() {
+      this.choiceRows.push({
+        id: this.nextChoiceRowId++,
+        imageurl: "",
+        value: "",
+      })
+    },
+    addNewChoiceRadio: function() {
+      this.choiceRowsRadio.push({
+        id: this.nextChoiceRowRadioId++,
+        imageurl: "",
+        value: "",
+      })
+    },
+    removeChoiceRowCheckbox: function(index, item) {
+      this.question.answerTypes.multipleChoice.checkboxArray.splice(index, 1)
+      for (let i = 0; i < this.preSelectedCheckBoxOptions.length; i++) {
+        if (item.id === this.preSelectedCheckBoxOptions[i]) {
+          this.preSelectedCheckBoxOptions.splice(i, 1)
+        }
+      }
+    },
+    removeChoiceRowRadio: function(index, item) {
+      this.question.answerTypes.multipleChoice.radioArray.splice(index, 1)
+      for (let i = 0; i < this.preSelectedRadioOptions.length; i++) {
+        if (item.id === this.preSelectedRadioOptions[i]) {
+          this.preSelectedRadioOptions.splice(i, 1)
+        }
+      }
+    },
+    updateOrderingCheckBoxArray(arr) {
+      this.question.answerTypes.multipleChoice.checkboxArray = arr
+    },
+    updateOrderingRadioArray(arr) {
+      this.question.answerTypes.multipleChoice.radioArray = arr
```

Comment:
If you only need the multipleChoice property of the question, it would be cleaner if you pass that in as a prop instead of the whole question. Otherwise, adding a computed property to return the multipleChoice will make all these lines much shorter and easier to read.